### ISSUE TITLE
Availability on F-Droid

### ISSUE THREAD

**Issue by onyxdhakal** [aniyomiorg/aniyomi#60]
Will this app make its way to fdroid as its going to be easier to track updates?

**Comment by jmir1** [comment]
no, because the original tachiyomi got deleted off of f-droid and i don't think i'll have any more luck with this version staying on there.

**Comment by ahmed-tasaly** [comment]
What about IzzyOnDroid or your own repo ?

**Comment by Raif1** [comment]
> no, because the original tachiyomi got deleted off of f-droid and i don't think i'll have any more luck with this version staying on there.

We do have tachiyomi in fdroid now . Do you plan to put it on fdroid?

**Comment by Thewisem** [comment]
Hmm I think, Izzy would be more than happy to set you up if Fdroid doesn't want you.

**Comment by momobobe** [comment]
> Hmm I think, Izzy would be more than happy to set you up if Fdroid doesn't want you.

Actually not because of the trackers. How do you think @izzysoft?

@jmir1 would you like to reconsider it? As far as I see, tachiyomi has been on F-droid since 2016 (not sure if it was removed sometime, but it's for sure there in the recent two years), then there's no need for your worry. The only issue is to remove trackers and other non-free (usually nonessential) components. If you may open your mind, their team are very willing to help you deal with the details. Thanks for your attention.

**Comment by IzzySoft** [comment]
Unfortunately just the size alone already argues against my repo. Even the per-ABI build is already at 40M+ – while the per-app limit in my repo is 30M. Running it through my library scanner (which btw is also used by F-Droid) gives a second (and stronger) "No":

```
Offending libs:
---------------
* Crashlytics (/com/crashlytics): NonFreeDep,Tracking
* JUnrar (/com/github/junrar): NonFreeDep
* Firebase Data Transport (/com/google/android/datatransport): NonFreeNet
* Google Mobile Services (/com/google/android/gms): NonFreeDep
* Firebase (/com/google/firebase): NonFreeNet,NonFreeDep
* Firebase Analytics (/com/google/firebase/analytics): NonFreeDep,Tracking

6 offenders.
```

Up to 2 or 3 it's borderline, but 6 is clearly too much even for my repo.

**Comment by jmir1** [comment]
About this:

>```
> Offending libs:
> ---------------
> * Crashlytics (/com/crashlytics): NonFreeDep,Tracking
> * JUnrar (/com/github/junrar): NonFreeDep
> * Firebase Data Transport (/com/google/android/datatransport): NonFreeNet
> * Google Mobile Services (/com/google/android/gms): NonFreeDep
> * Firebase (/com/google/firebase): NonFreeNet,NonFreeDep
> * Firebase Analytics (/com/google/firebase/analytics): NonFreeDep,Tracking
> 
> 6 offenders.

All but Junrar can be removed without any issues for a special tracker-free build.
I can't change much about the filesize though, so I guess we have better chances in the official F-Droid repo than in Izzydroid's.


@momobobe I just closed this issue initially because Tachi had some issues with a complaint by Webtoons that got the app removed for a while.
Now that Tachi is back on F-Droid, there isn't much that speaks against making an application to get Aniyomi on there.
Except that I don't have a lot of time right now so it's not high on my agenda.

But maybe soon :)

**Comment by IzzySoft** [comment]
@jmir1 that would be a huge step towards making the app "real foss" (i.e. removing the "tainting"). For my repo, junrar wouldn't be a real issue – so if it weren't for the size… Maybe let's see what effect removing the other 5 libs might have on that, though I doubt it will make more than 2M size decrease.

**Comment by RokeJulianLockhart** [comment]
@IzzySoft, why is the size of the file problematic? 40 MiB does not appear to be large enough to be problematic for most modern storage devices.

**Comment by IzzySoft** [comment]
Because a.o. …

* the repo runs on my personal storage
* the repo currently holds close to 900 apps
* a limit of 30M means 18G already
* there are limits on some APIs I use were I have to work-around for bigger apps

I have to set limits to keep it manageable. And remember, not everyone can afford a "modern device". Try running a 40M app on an "Android GO" device (gets already unbearable at around 25M in many cases – I can tell as I used such a device for reviews at F-Droid). So as a side-effect, this might be an incentive to slim an app where possible :wink:

**Comment by marc0x1** [comment]
Can a fully foss version be released on github?

**Comment by RokeJulianLockhart** [comment]
@spazziale, surely that's a different issue?

**Comment by IzzySoft** [comment]
@rokejulianlockhart not really – depends on you view-point. Having such APK available here would allow F-Droid to go for reproducible builds (if the builds are reproducible). On success, the APK signed by the developers would be published at -Droid, thus making cross-updates possible. So the two are connected.

**Comment by jmir1** [comment]
> Can a fully foss version be released on github?

By that I suppose you mean a version without Firebase/Google Analytics?
It's definitely possible, I will remember this and might implement it soon.
If I also managed to remove Junrar, the app would be ready for F-Droid inclusion then.

**Comment by IzzySoft** [comment]
> By that I suppose you mean a version without Firebase/Google Analytics?

Yes, definitely without those two and without Junrar (for F-Droid.org).

> It's definitely possible, I will remember this and might implement it soon.

Great, looking forward to that!

> If I also managed to remove Junrar, the app would be ready for F-Droid inclusion then.

Yupp. At that point you can start an [RFP](https://gitlab.com/fdroid/rfp/-/issues/) (Request For Packaging).

**Comment by IzzySoft** [comment]
That's a good start (added to my [list of repositories](https://android.izzysoft.de/articles/named/list-of-fdroid-repos). But it requires that people know about the repo and add it to their client. It's not the same as having the app on F-Droid.org :wink:

**Comment by Gamer750** [comment]
> Can a fully foss version be released on github?

It would be good

**Comment by rktfier** [comment]
> https://fdroid.aniyomi.org/

This mostly fixes the issue, since some other apps also have their own repositories set up so they don't have to go through all the steps of getting their apps accepted and updated on a separate repository.

Of course it would be another thing to see Aniyomi pruned of all the tracking, so still hoping to see the app on fdroid.org

It may also be possible to host extensions via the Aniyomi repository but i don't know a lot about Android development so it's up to the maintainers to figure out.

**Comment by jmir1** [comment]
> Of course it would be another thing to see Aniyomi pruned of all the tracking, so still hoping to see the app on fdroid.org
 
The preview version already has no analytics/tracking, just our private crash reporting that you can disable.

**Comment by rktfier** [comment]
> > Of course it would be another thing to see Aniyomi pruned of all the tracking, so still hoping to see the app on fdroid.org
> 
> The preview version already has no analytics/tracking, just our private crash reporting that you can disable.

That's nice to hear, but does that mean Aniyomi Preview versions can report crashes without an interaction from the user?

Also i'd like to think of the preview versions as what makes Aniyomi really shine out due to the amounts of features, so would it be considerable for only the preview to be on Fdroid repository?

**Comment by jmir1** [comment]
> > > Of course it would be another thing to see Aniyomi pruned of all the tracking, so still hoping to see the app on fdroid.org
> > 
> > The preview version already has no analytics/tracking, just our private crash reporting that you can disable.
> 
> That's nice to hear, but does that mean Aniyomi Preview versions can report crashes without an interaction from the user?
> 
> Also i'd like to think of the preview versions as what makes Aniyomi really shine out due to the amounts of features, so would it be considerable for only the preview to be on Fdroid repository? 

Yes the crash reporting is done automatically.
You can choose to install the preview or the stable version from the fdroid repo. The stable version is obviously very outdated at the moment so the preview is recommended.

**Comment by IzzySoft** [comment]
@jmir1 

> Yes the crash reporting is done automatically.

That would still mean the `Tracking` anti-feature at F-Droid. Better make that "opt-in" – ideally presenting crash reports to the users and let them confirm they should be sent after having the chance to see **what** would be sent (that is what e.g. F-Droid does via [ACRA](https://github.com/ACRA/acra); I didn't check what you are using as the last release here is from February, so that APK would not reflect your current builds).

**Comment by jmir1** [comment]
> @jmir1 
> 
> > Yes the crash reporting is done automatically.
> 
> That would still mean the `Tracking` anti-feature at F-Droid. Better make that "opt-in" – ideally presenting crash reports to the users and let them confirm they should be sent after having the chance to see **what** would be sent (that is what e.g. F-Droid does via [ACRA](https://github.com/ACRA/acra); I didn't check what you are using as the last release here is from February, so that APK would not reflect your current builds).

Yes we use ACRA.
Thanks for the suggestion, I think we can make it so the user can choose to enable automatic crash reports when they encounter a crash for the first time. That sort of screen could also detail what metadata is being sent.

**Comment by IzzySoft** [comment]
Thanks! Consider having an option in settings or so should someone change their minds later. F-Droid asks for each crash report separately AFAIK ("AFAIK" as I've never saw a crash of their app on any of my devices, so I cannot say this first-hand). Considering that crashes (hopefully :see_no_evil:) are not **that** frequent, this might also be an option.

**Comment by joshcangit** [comment]
> https://fdroid.aniyomi.org/

Can't add that repository to F-Droid.

It ends up showing as invalid repository and with this in gray below. `org.fdroid.download.NotFoundException`.

**Comment by fakebhai** [comment]
> > https://fdroid.aniyomi.org/
> 
> Can't add that repository to F-Droid.
> 
> It ends up showing as invalid repository and with this in gray below. `org.fdroid.download.NotFoundException`.

![Screenshot_2024_0307_174224.jpg](https://github.com/aniyomiorg/aniyomi/assets/101493467/0e53ea81-e149-477a-a941-6cce077d8b0c)

seems to be working just fine

**Comment by jmir1** [comment]
> > [fdroid.aniyomi.org](https://fdroid.aniyomi.org/)
> 
> Can't add that repository to F-Droid.
> 
> It ends up showing as invalid repository and with this in gray below. `org.fdroid.download.NotFoundException`.

I'm not sure why that's happening but at the moment, I'm having some issues with F-Droid anyway because the server I'm hosting it on ran out of storage space :(

**Comment by MicroBeast** [comment]
> > > [fdroid.aniyomi.org](https://fdroid.aniyomi.org/)
> > 
> > 
> > Can't add that repository to F-Droid.
> > It ends up showing as invalid repository and with this in gray below. `org.fdroid.download.NotFoundException`.
> 
> I'm not sure why that's happening but at the moment, I'm having some issues with F-Droid anyway because the server I'm hosting it on ran out of storage space :(

Maybe need to inform user that repo far outdated (currently) compared to Github's releases?

**Comment by joshcangit** [comment]
If it updates from within the app, then that fdroid repository maybe should be removed.
Especially since it is outdated.

**Comment by rktfier** [comment]
I don't agree. It allows users to download the initial app through their preferred store instead of finding the repository and downloading it through releases then going through a manual install (unlike with Fdroid, search-install with shizuku)

**Comment by shuvashish76** [comment]
@jmir1
> because the original tachiyomi got deleted off of f-droid

https://f-droid.org/packages/eu.kanade.tachiyomi/
No it's not yet removed from F-Droid, as long as it works for the users.

Requested at F-Droid: https://gitlab.com/fdroid/rfp/-/issues/2722

**Comment by linsui** [comment]
I opened a PR for you. :) https://github.com/aniyomiorg/aniyomi/issues/60

**Comment by Fellow-foss-lover** [comment]
Any update regarding it..? It would be really good if it could be made available in f droid

**Comment by linsui** [comment]
There is no new release. Mihon will be available in F-Droid and there is already a fork of Mihon in F-Droid.

**Comment by shuvashish76** [comment]
@linsui > Mihon will be available in F-Droid and there is already a fork of Mihon in F-Droid.

Mihon or Aniyomi _(this repo)_? Confused with too many manga apps !?

**Comment by linsui** [comment]
Mihon. And its fork kahon has been published in F-Droid.

**Comment by RokeJulianLockhart** [comment]
@DJ-Dingleberry, why the closure (as complete)? I ask because [`search.f-droid.org/?q=Aniyomi&lang=en`](https://search.f-droid.org/?q=Aniyomi&lang=en) returns 0 results.

**Comment by illiania100** [comment]
Why close, aniyomi still isn't in fdroid repo nor izzyondroid repo
